client/plots/disco_new/mapper/CnvArcsMapper.ts

```diff
@@ -0,0 +1,190 @@
+        this.gainCapped = this.settings.cnv.capping
+        this.lossCapped = -1 * this.settings.cnv.capping
+        this.lossOnly = cnvMaxValue <= 0
+        this.gainOnly = cnvMinValue >= 0
+
+        this.onePxArcAngle = 1 / (settings.rings.cnvInnerRadius)
+
+        const gain = new CnvLegend("Max", (cnvMaxValue > 0)?CnvType.Gain: CnvType.Loss, this.getColor(cnvMaxValue), cnvMaxValue)
+        const loss = new CnvLegend("Min", (cnvMinValue > 0)?CnvType.Gain: CnvType.Loss, this.getColor(cnvMinValue), cnvMinValue)
+        const cap = new CnvLegend("Capping", CnvType.Loss, this.getColor((cnvMinValue > 0)? cnvMinValue : cnvMaxValue), this.settings.cnv.capping)
+
+        this.cnvClassMap.set(CnvType.Gain, gain)
+        this.cnvClassMap.set(CnvType.Loss, loss)
+        this.cnvClassMap.set(CnvType.Cap, cap)
+
+        this.maxAbsValue = Math.max(Math.abs(this.capMaxValue(cnvMinValue)), Math.abs(this.capMaxValue(cnvMaxValue)))
+    }
+
+    map(arcData: Array<Data>): Array<CnvArc> {
+        const arcs: Array<CnvArc> = []
+
+        arcData.forEach(data => {
+            let startAngle = this.calculateStartAngle(data)
+            let endAngle = this.calculateEndAngle(data)
+
+            if (endAngle - startAngle < this.onePxArcAngle) {
+                const restAngle = this.onePxArcAngle - (endAngle - startAngle)
+                startAngle = startAngle - restAngle / 2
+                endAngle = startAngle + restAngle / 2
+            }
+
+            const innerRadius = this.calculateInnerRadius(data)
+
+            const outerRadius = this.calculateOuterRadius(data)
+
+            const color = this.getColor(data.value)
+
+            const arc = new CnvArc(startAngle,
+                endAngle,
+                innerRadius,
+                outerRadius,
+                color,
+                data.gene,
+                data.chr,
+                data.start,
+                data.stop,
+                data.value,
+                this.cnvUnit
+            )
+
+            arcs.push(arc)
+        })
+
+        return arcs
+    }
+
+    calculateStartAngle(data: Data) {
+        const index = this.reference.chromosomesOrder.indexOf(data.chr)
+        const chromosome = this.reference.chromosomes[index]
+        return chromosome.startAngle + ((chromosome.endAngle - chromosome.startAngle) * (Number(data.start) / chromosome.size));
+    }
+
+    private calculateEndAngle(data: Data) {
+        const index = this.reference.chromosomesOrder.indexOf(data.chr)
+        const chromosome = this.reference.chromosomes[index]
+        return chromosome.startAngle + ((chromosome.endAngle - chromosome.startAngle) * (Number(data.stop) / chromosome.size));
+    }
+
+    getColor(value) {
+
+        const lossColor = this.settings.cnv.lossColor
+        const ampColor = this.settings.cnv.ampColor
+        const cappedLossColor = this.settings.cnv.cappedLossColor
+        const cappedAmpColor = this.settings.cnv.cappedAmpColor
+
```

Comment:
@aacic The variable reassignments above are not needed, since they are only used once within this function, so there is no readability gained by having this extra layer of indirection. For example, just return the settings directly
```js
         if (value < this.lossCapped) {
            return this.settings.cnv.cappedLossColor
        } ...
```

If a setting is used in more than one place within the same function, then it may be simpler to assign the variable alias to the parent config, like this
```js
        const s = this.settings.cnv
        
        if (value < this.lossCapped) {
            return s.cappedLossColor
        } ...
``